platform/mellanox/docker-syncd-mlnx/ecmp_calculator/ecmp_calc.py

```diff
@@ -289,26 +292,30 @@ def print_egress_port(self):
     def is_port_bind_to_user_vrf(self, port_type, port, vlan_id = 0):
         if port_type == PORT:
             # INTF_TABLE:Ethernet0
-            entry = '{}:{}'.format(INTF_TABLE, port) 
+            entry = '{}'.format(port)
         elif port_type == VPORT:
             # INTF_TABLE:Ethernet0.300
-            entry = '{}:{}.{}'.format(INTF_TABLE, port, vlan_id)     
+            entry = '{}.{}'.format(port, vlan_id)
         elif port_type == VLAN:
             # INTF_TABLE:Vlan300
-            entry = '{}:Vlan{}'.format(INTF_TABLE, vlan_id)
+            entry = 'Vlan{}'.format(vlan_id)
+
+        vrf_table = Table(self.appl_db, INTF_TABLE)
+        (_, port_vrf) = vrf_table.hget(entry, 'vrf_name')
 
-        port_vrf = exec_cmd(['/usr/bin/redis-cli', '-n', '0', 'hget', entry, 'vrf_name'])
         if self.user_vrf == port_vrf.strip():
             return True
         
         return False
     
     # Get port-channel name for given port-channel member port
-    def get_port_channel_name(self, port):        
-        query_output = exec_cmd(['/usr/bin/redis-cli', '-n', '0', 'keys','*LAG_MEMBER_TABLE*'])
-        for line in query_output.split('\n'):
-            if str(port) in line:
-                port_channel = line.split(':')[PORT_CHANNEL_IDX]
+    def get_port_channel_name(self, port):
+        lag_member_table = Table(self.appl_db, LAG_MEMBER_TABLE)
+        lag_member_table_keys = lag_member_table.getKeys()
+
+        for key in lag_member_table_keys:
```

Comment:
Also check here if you can remove the iteration

Reply:
In this case I can not remove the iteration.

lag_member_table_keys looks like:
('PortChannel300:Ethernet128', 'PortChannel200:Ethernet0', 'PortChannel200:Ethernet124')

I must iterate the keys in order to see if key contain the interface I am looking for.